compiler/quilt/tools/build.py

```diff
@@ -63,7 +63,10 @@ def _path_hash(path, transform, kwargs):
     return digest_string(srcinfo)
 
 def _is_internal_node(node):
-    is_leaf = not node or node.get(RESERVED['file'])
+    try:
+        is_leaf = not node or node.get(RESERVED['file'])
+    except AttributeError:
+        raise BuildException("Invalid yml for quilt: expected node data, but got {!r} instead".format(node))
```

Comment:
~~for quilt~~
~~but~~

Reply:
So we want: `Invalid syntax for <FILE>: expected node data, got {!r}".format(node))`